builder/src/main/java/cz/xtf/builder/db/DB2.java

```diff
@@ -0,0 +1,223 @@
+    private static final String DEFAULT_DB2_USER_ENV_VAR = "DB2INSTANCE";
+    private static final String DEFAULT_DB2_PASS_ENV_VAR = "DB2INST1_PASSWORD";
+    private static final String DEFAULT_DB2_DATABASE_ENV_VAR = "DBNAME";
+
+    private Map<String, String> vars;
+    private List<String> args;
+    private String serviceAccount;
+
+    public DB2(Builder builder) {
+        super(
+                builder.symbolicName,
+                builder.dataDir,
+                builder.pvc,
+                builder.username,
+                builder.password,
+                builder.dbName,
+                builder.configureEnvironment,
+                builder.withLivenessProbe,
+                builder.withReadinessProbe,
+                builder.withStartupProbe,
+                builder.deploymentConfigName,
+                builder.envVarPrefix);
+        this.vars = builder.vars;
+        this.args = builder.args;
+        this.serviceAccount = builder.serviceAccount;
+    }
+
+    @Override
+    public String getImageName() {
+        return Image.resolve("db2").getUrl();
+    }
+
+    @Override
+    public int getPort() {
+        return 50000;
+    }
+
+    protected ProbeSettings getProbeSettings() {
+        return new ProbeSettings(120,
+                String.valueOf(getPort()),
+                5,
+                // Adding SQL query in liveness probe causes failure.
+                String.format("su - %s -c 'db2 -x \"connect to %s\"'",
+                        getUsername(), getDbName()),
+                120,
+                String.format("su - %s -c 'db2 -x connect to %s'",
+                        getUsername(), getDbName()),
+                10,
+                10);
+    }
+
+    @Override
+    public Map<String, String> getImageVariables() {
+        Map<String, String> vars;
+        vars = new HashMap<>();
+        vars.put("LICENSE", "accept"); // Required for DB2 container to start
+        vars.put(DEFAULT_DB2_DATABASE_ENV_VAR, getDbName());
+        vars.put(DEFAULT_DB2_USER_ENV_VAR, getUsername());
+        vars.put(DEFAULT_DB2_PASS_ENV_VAR, getPassword());
+        vars.putAll(this.vars);
+        return vars;
+    }
+
+    @Override
+    public String toString() {
+        return "DB2";
+    }
+
+    @Override
+    protected String getJDBCConnectionStringPattern() {
+        return "jdbc:db2://%s:%s/%s";
+    }
+
+    @Override
+    public List<String> getImageArgs() {
+        return args;
+    }
+
+    @Override
+    public String getServiceAccount() {
+        return serviceAccount;
+    }
+
+    @Override
+    boolean requiresInitContainer() {
+        return true;
+    }
+
+    @Override
+    boolean requiresPrivileged() {
+        return true;
+    }
+
+    @Override
+    void configureInitContainer(ContainerBuilder initContainer) {
+        initContainer.fromImage(getImageName()) // reuse the same image as the main container
+                .addCommand("sh", "-c",
```

Comment:
This solution may feel hacky, but when attempting scale down and scale up DB2 image ends in failure loopback, due to existing data being present. This effectively removes whole existing database.

It may be possible to pinpoint this to some config files, but I didn't bother with this (yet).

Reply:
well we need to sort this out, or if not possible, we can use this image without persistent storage and save us the trouble with deleting files? but it is quite odd for database to be unable to use persistent storage